src/main/java/com/kramekk1/medicalClinic/PatientController.java

```diff
@@ -0,0 +1,41 @@
+package com.kramekk1.medicalClinic;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/patients")
+public class PatientController {
+
+    private final PatientRepository patientRepository;
+    private final PatientService patientService;
+
+    @GetMapping("/all")
+    public List<Patient> getPatients() {
+        return patientRepository.getPatients();
+    }
+    @GetMapping("/{email}")
+    public Optional<Patient> getPatientByEmail(@PathVariable String email) {
+        return patientRepository.findPatientByEmail(email);
+    }
+
+    @PostMapping("/add")
+    public String addPatient(@RequestBody Patient patient) {
+        patientRepository.addPatient(patient);
+        return "Dodano pomyslnie";
+    }
+    @PutMapping("/{email}")
+    public String updatePatient(@PathVariable String email, @RequestBody Patient newPatient) {
+        patientService.editPatientByEmail(email, newPatient);
+        return "Zaktualizowano pacjenta";
+    }
+    @DeleteMapping("/{email}")
+    public String deletePatient(@PathVariable String email) {
```

Comment:
Nie zwracajmy stringa mozemy zwrocic albo pacjenta ktorego usunelismy albo voida -> puste body
Jak zwracamy puste body to mamy lepszy status HTTP niz OK -> NO_CONTENT